tests/integration_tests/datasources/test_oracle_ipv6.py

```diff
@@ -0,0 +1,186 @@
+import time
```

Comment:
This test is a great start, but I think we also need some testing beyond IMDS reachability. In particular:
* No long waits or timeouts when fetching the data
* Network rendered properly
* No errors or warnings in the logs

I think it's important to verify these things on both native and paravirtualized instances. I also realize some of this isn't really viable using our current integration testing tools and that this functionality isn't widely available yet, but I do think that means we also need some manual checks in the meantime.